libr/anal/fcn.c

```diff
@@ -265,6 +265,64 @@ static ut64 search_reg_val(RAnal *anal, ut8 *buf, ut64 len, ut64 addr, char *reg
 
 #define gotoBeach(x) ret=x;goto beach;
 #define gotoBeachRet() goto beach;
+
+R_API void fill_args (RAnal *anal, RAnalFunction *fcn, RAnalOp *op) {
+	char* varname;
+	switch (op->stackop) {
+	case R_ANAL_STACK_NULL:
+	case R_ANAL_STACK_NOP:
+	case R_ANAL_STACK_ALIGN:
+	case R_ANAL_STACK_INC: {
+		//When everyone is busy I have to figure my way out
+		//nothing is impossible
+		char *esil_buf = strdup (op->esil.buf);
+		char *ptr_end = NULL;
+		if (!(ptr_end = strstr (esil_buf, ",ebp,+,[4],"))) {
+			goto end;
+		}
+		*ptr_end = 0;
+		char *addr = ptr_end;
+		while((*addr != '0' || *(addr+1) != 'x') &&addr >= esil_buf && *addr != ',' ) {
+			addr--;
+		}
+		if (strncmp(addr, "0x", 2)) {
+			goto end;
+		}
+		st64 ptr;
+		sscanf (addr, "%llx", &ptr);
```

Comment:
ptr = r_num_get(NULL, addr)
> On 22 Apr 2016, at 11:28, Ahmed Mohamed Abd El-MAwgood <notifications@github.com> wrote:
> 
> In libr/anal/fcn.c <https://github.com/radare/radare2/pull/4676#discussion_r60712990>:
> 
> > +		//nothing is impossible
> > +		char *esil_buf = strdup (op->esil.buf);
> > +		char *ptr_end = NULL;
> > +		if (!(ptr_end = strstr (esil_buf, ",ebp,+,[4],"))) {
> > +			goto end;
> > +		}
> > +		*ptr_end = 0;
> > +		char *addr = ptr_end;
> > +		while((*addr != '0' || *(addr+1) != 'x') &&addr >= esil_buf && *addr != ',' ) {
> > +			addr--;
> > +		}
> > +		if (strncmp(addr, "0x", 2)) {
> > +			goto end;
> > +		}
> > +		st64 ptr;
> > +		sscanf (addr, "%llx", &ptr);
> cant have access to core..
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/radare/radare2/pull/4676/files/af0d81f0f8d76289e2b2d9fd063c047660e44345#r60712990>